### PR TITLE
satellite/metainfo: update partner id from request

### PR DESCRIPTION
**Pull request by aligeti**
What: 
Adds  the partnerid to the bucketsDB, when a new bucket is created or updats an already existing bucket if partner id exists in the APIkey info, used to access(create) a new bucket.
Why:
OSPP integration testing

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/metainfo/metainfo.go [hunk 1]
```diff
@@ -645,15 +645,10 @@ func (endpoint *Endpoint) CreateBucket(ctx context.Context, req *pb.BucketCreate
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	// checks if bucket exists before updates it or makes a new entry
+	// checks if bucket exists before updates it with partner id if it exits or makes a new entry
+	partnerID := keyInfo.PartnerID
 	bucket, err := endpoint.metainfo.GetBucket(ctx, req.GetName(), keyInfo.ProjectID)
 	if err == nil {
-		var partnerID uuid.UUID
```
Comment (VinozzZ) [inline satellite/metainfo/metainfo.go]: Do we still need this logic for the app-connector?

File: satellite/metainfo/metainfo.go [hunk 2]
```diff
@@ -645,15 +645,10 @@ func (endpoint *Endpoint) CreateBucket(ctx context.Context, req *pb.BucketCreate
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	// checks if bucket exists before updates it or makes a new entry
+	// checks if bucket exists before updates it with partner id if it exits or makes a new entry
+	partnerID := keyInfo.PartnerID
 	bucket, err := endpoint.metainfo.GetBucket(ctx, req.GetName(), keyInfo.ProjectID)
 	if err == nil {
```
Comment (coyle) [inline satellite/metainfo/metainfo.go]: This is actually what @egonelbre 's comments came from a few weeks ago. We should be pulling the partnerID from a user-agent string not the request object.

File: satellite/metainfo/metainfo.go (hunk 1 above)
Comment (aligeti) [reply satellite/metainfo/metainfo.go]: updated the logic

File: satellite/metainfo/metainfo.go (hunk 2 above)
Comment (aligeti) [reply satellite/metainfo/metainfo.go]: got it..

**Comment by aligeti** [comment]
> LGTM, other than the comment from Dennis. Also is there tests for this?

there are test cases written in place... I will double check it again.

File: satellite/metainfo/metainfo.go (hunk 2 above)
Comment (mniewrzal) [reply satellite/metainfo/metainfo.go]: [How to get UserAgent on server side](https://github.com/grpc/grpc-go/issues/1100)

**Comment by aligeti** [comment]
> LGTM, other than the comment from Dennis. Also is there tests for this?

Line #649 keyInfo structure has partnerID that is extracted thru ctx.Value(ctx). so the partner ID is extracted thru user-agent.

File: satellite/metainfo/metainfo.go (hunk 2 above)
Comment (coyle) [reply satellite/metainfo/metainfo.go]: Let's do this in this PR

File: satellite/metainfo/validation.go [hunk 3]
```diff
@@ -371,3 +381,29 @@ func (endpoint *Endpoint) validatePieceHash(ctx context.Context, piece *pb.Remot
...
+func (endpoint *Endpoint) getUserAgent(ctx context.Context) (_ *uuid.UUID, err error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		if userAgent, ok := md["user-agent"]; ok {
+			if len(userAgent) == 1 {
+				ua := useragent.Parse(userAgent[0])
+				if ua == nil {
+					endpoint.log.Sugar().Info("invalid user-agent found %s", userAgent[0])
+					return nil, status.Errorf(codes.InvalidArgument, err.Error())
+				}
+				partnerID, err := rewards.GetPartnerID(ua.Name)
```
Comment (VinozzZ) [inline satellite/metainfo/validation.go]: Should we also check if the `ua.PartnerID` is equal to the `partnerID` that we get from the satellite?

Comment (VinozzZ) [reply satellite/metainfo/validation.go]: After the meeting today, we will not validate the partner ID here and only validate it during billing

**Pull request closed without merging**